server/routes/propertyVideo.routes.js

```diff
@@ -0,0 +1,165 @@
+ *      '200':
+ *        description: Video added
+ *      '404':
+ *        description: Property not found
+ *      '500':
+ *       description: Internal server error
+ */
+router.post(
+  '/',
+  authenticate,
+  isVendor,
+  schemaValidation(addPropertyVideoSchema),
+  PropertyVideoController.addVideo,
+);
+
+/**
+ * @swagger
+ * /property-video:
+ *   put:
+ *     tags:
+ *       - PropertyVideo
+ *     description: Updates a video for a property
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/PropertyVideo'
+ *      description: Updates a video for a property
+ *     responses:
+ *      '200':
+ *        description: Video updated
+ *      '400':
+ *        description: Error updating video
+ *      '500':
+ *       description: Internal server error
+ */
+router.put(
+  '/',
+  authenticate,
+  isVendor,
+  schemaValidation(updatePropertyVideoSchema),
+  PropertyVideoController.updateVideo,
+);
+
+/**
+ * @swagger
+ * /property-video/:id/approve:
+ *   put:
+ *     tags:
+ *       - PropertyVideo
+ *     description: Approve a video
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/PropertyVideo'
+ *      description: Approve a video
+ *     responses:
+ *      '200':
+ *        description: Video approved
+ *      '400':
+ *        description: Error approving video
+ *      '500':
+ *       description: Internal server error
+ */
+router.put(
+  '/:id/approve',
+  authenticate,
+  isAdmin,
+  hasValidObjectId,
+  PropertyVideoController.approveVideo,
+);
+
+/**
+ * @swagger
+ * /property-video/:id:
+ *   delete:
+ *     tags:
+ *       - PropertyVideo
+ *     description: Deletes a video in a property's videos
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/PropertyVideo'
+ *      description: Deletes a video in a property's videos
+ *     responses:
+ *      '200':
+ *        description: Video deleted
+ *      '500':
+ *       description: Internal server error
+ */
+router.delete(
+  '/:id',
+  authenticate,
+  hasValidObjectId,
+  isVendor,
+  PropertyVideoController.deleteVideo,
+);
+
+/**
+ * @swagger
+ * /property-video/:id/all:
+ *   get:
+ *     tags:
+ *       - PropertyVideo
+ *     description: Get all videos for a property
+ *     produces:
+ *       - application/json
+ *     requestBody:
+ *      content:
+ *        application/json:
+ *          schema:
+ *            $ref: '#/components/schemas/PropertyVideo'
+ *      description: Get all videos for a property
+ *     responses:
+ *      '200':
+ *        description: returns videos
+ *      '500':
+ *       description: Internal server error
+ */
+router.get(
+  '/:id/all',
```

Comment:
```suggestion
  '/:propertyId/all',
```

In this scenario, we need to use the `propertyId` instead of `id` to differentiate that the id in question is not the video id but the property Id

Reply:
Using `:propertyId` will prevent Joi from validating the id. The description will be stated in the description.